modules/meta.analysis/inst/magic/initialize_planting.R

```diff
@@ -1,24 +1,138 @@
-library(dplyr)
+source("get_stats.R")
+source("get_trait_value.R")
+source("get_fallback_class_val.R")
+source("get_fallback_type_val.R")
 
-initialize_planting <- function(species_name) {
+initialize_planting <- function(UniqueID, Date, CLASS_SUBCLASS, LAI) {
   
-  species_stats <- data.frame(
-    get_stats(planting_df, "OrigValueStr", "TraitID", "SpeciesName", species_name)
-  )
-  return(species_stats)
+  if (!is.numeric(UniqueID)||!is.character(Date)||!is.character(CLASS_SUBCLASS)||!is.numeric(LAI)){
+    stop ("Incorrect types!")
+  }
   
-  #calculate for each trait 
+  group_name <- get_crop_name(CLASS_SUBCLASS)
+  
+  LAI = LAI * 10000
+  
+  #val_601 <- get_fallback_value(group_name, trait_id = 601)
+  val_3441 <- get_fallback_value(group_name, trait_id = 3441)
```

Comment:
separate the calculation of parameters from there use in calculating pools. Parameters should all be precalculated into a simple static lookup table based on group_name